client/my-sites/earn/components/add-edit-coupon-modal/test/index.jsx

```diff
@@ -0,0 +1,243 @@
+import { unmountComponentAtNode } from 'react-dom';
+import Modal from 'react-modal';
+import {
+	getconnectedAccountDefaultCurrencyForSiteId,
+	getconnectedAccountMinimumCurrencyForSiteId,
+} from '../../../../../state/memberships/settings/selectors';
+import { renderWithProvider } from '../../../../../test-helpers/testing-library';
+import {
+	COUPON_DISCOUNT_TYPE_AMOUNT,
+	COUPON_DISCOUNT_TYPE_PERCENTAGE,
+	COUPON_DURATION_3_MONTHS,
+	COUPON_DURATION_FOREVER,
+} from '../../../memberships/constants';
+import RecurringPaymentsCouponAddEditModal from '../index';
+
+const closeDialog = jest.fn();
+
+describe( 'RecurringPaymentsCouponAddEditModal', () => {
+	let modalRoot;
+
+	const getRandomButton = () =>
+		screen.getByRole( 'button', { name: 'Generate random coupon code' } );
+	const getCodeTextbox = () =>
+		screen.getByRole( 'textbox', { name: 'Enter a custom coupon code' } );
+	const getDiscountTypeSelect = () => screen.getByRole( 'combobox', { name: 'Discount type' } );
+	const getAmountOption = () => screen.getByRole( 'option', { name: 'Amount' } );
+	const getDiscountPercentageInput = () =>
+		screen.getByRole( 'textbox', { name: 'Discount percentage' } );
+	const getDiscountValueInput = () => screen.getByRole( 'textbox', { name: 'Discount value' } );
+	const queryDiscountPercentageInput = () =>
+		screen.queryByRole( 'textbox', { name: 'Discount percentage' } );
+	const queryDiscountValueInput = () => screen.queryByRole( 'textbox', { name: 'Discount value' } );
+	const getLimitDurationToggle = () => screen.getByRole( 'checkbox', { name: 'Duration' } );
+	const getLimitDurationSelection = () =>
+		screen.getByRole( 'combobox', { name: 'Duration selection' } );
+	const getLimitCouponsToggle = () =>
+		screen.getByRole( 'checkbox', { name: 'Limit coupon to specific emails' } );
+	const getLimitCouponsTextInput = () =>
+		screen.getByRole( 'textbox', { name: 'Limit coupon to specific emails text input' } );
+
+	beforeEach( () => {
+		jest.clearAllMocks();
+
+		getconnectedAccountDefaultCurrencyForSiteId.mockImplementation(
+			mockGetConnectedAccountDefaultCurrencyForSiteId
+		);
+		getconnectedAccountMinimumCurrencyForSiteId.mockImplementation(
+			mockGetConnectedAccountMinimumCurrencyForSiteId
+		);
+
+		modalRoot = document.createElement( 'div' );
+		modalRoot.setAttribute( 'id', 'wpcom' );
+		document.body.appendChild( modalRoot );
+		Modal.setAppElement( modalRoot );
+	} );
+
+	afterEach( () => {
+		unmountComponentAtNode( modalRoot );
+		document.body.removeChild( modalRoot );
+		[ ...document.getElementsByClassName( 'ReactModalPortal' ) ].map( ( el ) => {
+			document.body.removeChild( el );
+		} );
+		modalRoot = null;
+	} );
+
+	test( 'should render empty form when no coupon is provided', () => {
+		const render = ( el, options ) => renderWithProvider( el, { ...options } );
+		render( <RecurringPaymentsCouponAddEditModal closeDialog={ closeDialog } coupon={ {} } /> );
+		const codeTextbox = getCodeTextbox();
```

Comment:
I usually don't recommend using those helper functions for queries because it makes tests less readable, and one has to scroll up and see what the query really is. This is underlined by the fact that we're assigning those to constants anyway, so that's one more reason to not have helper functions for queries. WDYT?